utilities/transactions/write_prepared_txn.cc

```diff
@@ -47,9 +57,21 @@ Status WritePreparedTxn::CommitWithoutPrepareInternal() {
 }
 
 Status WritePreparedTxn::CommitInternal() {
-  // TODO(myabandeh) Implement this
-  throw std::runtime_error("Commit not Implemented");
-  return Status::OK();
+  // We take the commit-time batch and append the Commit marker.
+  // The Memtable will ignore the Commit marker in non-recovery mode
+  WriteBatch* working_batch = GetCommitTimeWriteBatch();
+  WriteBatchInternal::MarkCommit(working_batch, name_);
+
+  // any operations appended to this working_batch will be ignored from WAL
+  working_batch->MarkWalTerminationPoint();
```

Comment:
working_batch = GetCommitTimeWriteBatch() meaning the writes done after prepare. This is an unusual way of using 2PC. Let me create an empty write batch instead and put a TODO to make sure we prevent such late writes.